meson.build

```diff
@@ -0,0 +1,76 @@
+project(
+  'libvfio-user',
+  'c',
+  version: '0.1',
+  license: 'BSD-3-Clause',
+  meson_version: '>= 0.53.0',
+  default_options: [
+    'buildtype=debugoptimized',
+    'c_std=gnu99',
+    'warning_level=2',
+  ],
+)
+
+opt_rpath = get_option('rpath')
+opt_tran_pipe = get_option('tran-pipe')
+opt_debug_logs = get_option('debug-logs')
+opt_valgrind = get_option('valgrind')
+opt_sanitizers = get_option('b_sanitize')
+
+if opt_valgrind and opt_sanitizers != 'none'
+  error('valgrind is incompatible with -Db_santizers')
+endif
+
+cc = meson.get_compiler('c')
+
+prefix = get_option('prefix')
+includedir = prefix / get_option('includedir')
+libdir = prefix / get_option('libdir')
+
+if prefix == '/usr' and not opt_rpath.enabled()
+  rpathdir = ''
+else
+  rpathdir = libdir
+endif
+
+thread_dep = dependency('threads')
+dl_dep = cc.find_library('dl', required: true)
+
+json_c_version = '0.11'
+json_c_dep = dependency('json-c', version: '>=' + json_c_version)
+
+openssl_version = ''
+openssl_dep = dependency('openssl', version: '>=' + openssl_version)
+
+cmocka_version = ''
+cmocka_dep = dependency('cmocka', version: '>=' + cmocka_version)
+
+
+pytest = find_program('pytest-3', required: false)
+flake8 = find_program('flake8', required: false)
+rstlint = find_program('restructuredtext-lint', required: false)
+valgrind = find_program('valgrind', required: opt_valgrind)
+
+
+common_cflags = [
+  '-D_GNU_SOURCE',
+]
+
+if opt_debug_logs
```

Comment:
this should get set by default unless we're in release mode I think?

Reply:
Yes, good point, will do so